python/hail/dataset.py

```diff
@@ -750,6 +750,63 @@ def annotate_variants_expr(self, condition):
         pargs = ['annotatevariants', 'expr', '-c', condition]
         return self.hc.run_command(self, pargs)
 
+    def annotate_variants_keytable(self, keytable, condition, vdsKey=None):
```

Comment:
I like the list approach as well, but I'm not sure how to square that with the KeyTable API. Since the key is always a structure and our users cannot predict the way we'll order the fields for equality, I'm not sure how to treat a list of expressions.